pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -252,15 +253,78 @@ func (g *generator) genFunctionUsings(x *model.FunctionCallExpression) []string
 	return []string{fmt.Sprintf("%s = Pulumi.%[1]s", pkg)}
 }
 
+// extractStringLiteral returns the string value of a string literal if a given expression is a single-part string
+// template expression.
+func extractStringLiteral(arg model.Expression) (string, bool) {
+	switch expr := arg.(type) {
+	case *model.TemplateExpression:
+		if arg.Type() != model.StringType || len(expr.Parts) != 1 {
+			return "", false
+		}
+		if lit, ok := expr.Parts[0].(*model.LiteralValueExpression); ok && lit.Type() == model.StringType {
+			return lit.Value.AsString(), true
+		}
+	}
+
+	return "", false
+}
+
+func (g *generator) genConvert(w io.Writer, expr *model.FunctionCallExpression) {
+	switch arg := expr.Args[0].(type) {
+	case *model.ObjectConsExpression:
+		g.genObjectConsExpression(w, arg, expr.Type())
+	default:
+		if targetType, ok := hcl2.GetSchemaForType(expr.Type().(model.Type)); ok {
```

Comment:
You shouldn't need the call to `GetSchemaForType` here. Instead, I'd use something like

```
switch model.ResolveOutputs(expr.Type()) {
case model.BoolType:
    ...
case model.IntType:
    ...
case model.NumberType:
    ...
case model.StringType:
    ...
}
```

Reply:
Good to know. I thought that only `GetSchemaForType` can disambiguate unions etc. I'll try on Monday!